src/adtypes_interface.jl

```diff
@@ -1,10 +1,34 @@
 #= This file implements the ADTypes interface for `AbstractSparsityDetector`s =#
+const DEFAULT_SET_TYPE = BitSet
+const DEFAULT_DICT_TYPE = Dict{eltype(DEFAULT_SET_TYPE), DEFAULT_SET_TYPE}
+const DEFAULT_SHARED_TYPE = NotShared
```

Comment:
Will Hessian sharing be public?

Reply:
Made it a documented boolean kwarg instead of revealing internal types.